internal/reader/onchain_prices_reader.go

```diff
@@ -4,40 +4,57 @@ import (
 	"context"
 	"fmt"
 	"math/big"
+	"time"
 
 	"github.com/smartcontractkit/chainlink-ccip/pkg/consts"
 	"github.com/smartcontractkit/chainlink-ccip/pluginconfig"
 
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/types"
 	ocr2types "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
 
-	commontypes "github.com/smartcontractkit/chainlink-common/pkg/types"
+	commontyps "github.com/smartcontractkit/chainlink-common/pkg/types"
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
 	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
 
 	"golang.org/x/sync/errgroup"
 )
 
-type TokenPrices interface {
-	// GetTokenPricesUSD returns the prices of the provided tokens in USD.
+type PriceReader interface {
+	// GetTokenFeedPricesUSD returns the prices of the provided tokens in USD normalized to e18. 1USD=1e18USD.
 	// The order of the returned prices corresponds to the order of the provided tokens.
-	GetTokenPricesUSD(ctx context.Context, tokens []ocr2types.Account) ([]*big.Int, error)
+	GetTokenFeedPricesUSD(ctx context.Context, tokens []ocr2types.Account) ([]*big.Int, error)
+
+	// GetFeeQuoterTokenUpdates returns the latest token prices from the FeeQuoter on the dest chain.
+	GetFeeQuoterTokenUpdates(
+		ctx context.Context,
+		tokens []ocr2types.Account,
+	) (map[ocr2types.Account]NumericalUpdate, error)
+}
+
+// NumericalUpdate On-chain struct `TimestampedPackedUint224`:
+// nolint:lll
+// https://github.com/smartcontractkit/chainlink/blob/2d77ff4623d0a0032533c89f32fc251400382455/contracts/src/v0.8/ccip/libraries/Internal.sol#L43-L47
+type NumericalUpdate struct {
```

Comment:
This type is duplicated across different files/packages

Reply:
Yes, it's going to be cyclic dependency if I have a new package for the shared types and use it in the internal/reader and commit as commit uses internal/reader.

Reply:
I don't think that's true. There is no cycle if you have the following organization:
* shared
* internal/reader imports shared
* commit imports internal/reader and shared
To get a cycle you need to have e.g commit import internal/reader which imports commit.

Reply:
fixing.